lookbook/docs/patterns/15-colors.md.erb

```diff
@@ -68,6 +68,151 @@ The following variables of Primer are overwritten with the `--accent-color`:
 
 When implementing colors in the application, we should ensure compliance with Primer's [accessibility standards](https://primer.style/foundations/color/accessibility).
 
+## Attributes colors
+
+We are following the same approach GuitHub is doing for their labels. For all configured colors, based on their hex codes and the selected mode, new values will be generated. The highlighting color values should be always visible and readable for:
+- Font color
+- Complete row highlighting
+- Card highlighting
+- Status selector
+- Highlighting via bubble
+
+There is a method that decompose the color hex code into the rgb and hsl values. While highlighting an attribute, this method should be called first:
+
+```
+def default_color_styles(hex)
+  color = ColorConversion::Color.new(hex)
+  rgb = color.rgb
+  hsl = color.hsl
+
+  "--color-r: #{rgb[:r]};
+   --color-g: #{rgb[:g]};
+   --color-b: #{rgb[:b]};
+   --color-h: #{hsl[:h]};
+   --color-s: #{hsl[:s]};
+   --color-l: #{hsl[:l]};
+   --perceived-lightness: calc( ((var(--color-r) * 0.2126) + (var(--color-g) * 0.7152) + (var(--color-b) * 0.0722)) / 255 );
+   --lightness-switch: max(0, min(calc((1/(var(--lightness-threshold) - var(--perceived-lightness)))), 1));"
+end
+```
+
+These two methods return default styles for defining brightness in dark and light modes:
+
+```
+def default_variables_dark
+  "--lightness-threshold: 0.6;
+   --background-alpha: 0.18;
+   --lighten-by: calc(((var(--lightness-threshold) - var(--perceived-lightness)) * 100) * var(--lightness-switch));"
+end
+
+def default_variables_light
+  "--lightness-threshold: 0.453;
+   --border-threshold: 0.75;
+   --border-alpha: max(0, min(calc((var(--perceived-lightness) - var(--border-threshold)) * 100), 1));"
+end
+```
+
+We have defined four methods that calculate and set correct styles for highlighting background and foreground colors in dark and light modes:
+
+```
+def highlighted_background_dark
+  "color: hsl(var(--color-h), calc(var(--color-s) * 1%), calc((var(--color-l) + var(--lighten-by)) * 1%)) !important;
+   background: rgba(var(--color-r), var(--color-g), var(--color-b), var(--background-alpha)) !important;
+   border: 1px solid hsl(var(--color-h), calc(var(--color-s) * 1%), calc((var(--color-l) + var(--lighten-by)) * 1%)) !important;"
+end
```

Comment:
You don't have to repeat all of these methods in the documentation. It will be outdated pretty quickly and is duplicating. Developers who intereseted in details can have a look at the code. It is better to explain exemplarily how it is done. The concrete formular to calculate values is not that important.